### DISCUSSION TITLE
API List page drop-down on rxjs.dev

### DISCUSSION THREAD

**Discussion by rendall** [ReactiveX/rxjs#6337] [General]
### Documentation Related To Component:

[API List page](https://rxjs.dev/api) "Type" drop-down 

### Description Of The Issue

This is both a request for comment and an offer to contribute. 

Currently, the "Type" drop-down on the [API List page](https://rxjs.dev/api)  allows the user to filter by these types: 

- Class
- Const
- Enum
- Function
- Interface
- Type Alias

Speaking personally, I find this to be unintuitive.  I have never needed to filter documentation in that way. I nearly always need to filter by "Operator" or "Observable".  

What do we think about adding library-specific terms to the API drop-down?  e.g.

- Operator
- Observable
- Observer
- Single
- Subject
- Scheduler
- etc? 

If the community comes to a consensus about this, I would be willing to code this and submit a pull-request. 

Thoughts?

(*edit*: I just noticed that either https://rxjs.dev is not a public repo, or I couldn't find it. Is a PR request on it possible?)

**Comment by benlesh** [comment]
This would require a lot of discussion. I'm not sure what a "Single" would be, for example... and we have to consider newbies that have no idea what an Operator even is.

**Comment by niklas-wortmann** [comment]
In general I like the idea. Newbies also doesn't care if a subject is a function, a class or a what ever. I think adding some filter could improve the usability. Also we could consider adding a filter for groups of operators, e.g. 'map operators', 'timing operators'... similar to the 'find your operator' widget. But also I'm not sure how often such a feature would be used. Because most of the time you are searching for a specific operator and just want to dig into that one, at least I am doing it this way :D Not sure if it's worth the effort

**Comment by rendall** [comment]
Agreed, and this is the motivation for my offer: I personally do not find the current categories intuitive or useful, and wonder if other library consumers have the same experience. I find myself trawling http://reactivex.io and then attempting to transliterate to https://rxjs.dev

Is the rxjs.dev website written *for* the library developers as opposed to *consumers*?  I'm not easily imagining a use case where library consumers of whatever experience would need to filter 'functions' or 'const', otherwise. Types of Observables, or various types of Operators, yes.

**Comment by jakovljevic-mladen** [comment]
Hey @rendall, I saw that you recently closed this issue of which I wasn't aware before.

I have a [WIP version](https://github.com/jakovljevic-mladen/rxjs/tree/add_operator_and_creationOperator_tag_definitions) of this request ([this commit specifically](https://github.com/jakovljevic-mladen/rxjs/commit/2d878ce0b194121b99688ff213888331798cd3b3)) in one of my fork branches for which I never created a PR because I had no enough time to test it at first and I later forgot about it. Once I saw this issue being closed recently, I wanted to let you know that we could potentially have a fix for it soon.

I'd like to reopen this issue again to force myself to create a PR for it. Though, it won't have all dropdown options that you suggested initially, but it will have the most common one: an option to filter operators which I guess would be the most useful one.
